FirmataMarshaller.cpp

```diff
@@ -168,6 +168,8 @@ const
   if ( (Stream *)NULL == FirmataStream ) { return; }
   FirmataStream->write(START_SYSEX);
   FirmataStream->write(REPORT_FIRMWARE);
+  FirmataStream->write(firmata::FIRMWARE_MAJOR_VERSION);
```

Comment:
This is a query for the firmware version so you don't know the version at this point.